src/main/java/org/folio/rest/helper/BudgetsHelper.java

```diff
@@ -29,17 +31,35 @@ public BudgetsHelper(HttpClientInterface httpClient, Map<String, String> okapiHe
   }
 
   public CompletableFuture<Budget> createBudget(Budget budget) {
-    return handleCreateRequest(resourcesPath(BUDGETS), budget).thenApply(budget::withId);
+    double allocatedValue = budget.getAllocated();
+    budget.setAllocated(0.0);
+    return handleCreateRequest(resourcesPath(BUDGETS), budget).thenCompose(budgetId -> {
+      if (allocatedValue > 0.0) {
+        return createAllocationTransaction(budget.withAllocated(allocatedValue)).thenApply(v -> budget.withId(budgetId));
+      }
+      return CompletableFuture.completedFuture(budget.withId(budgetId));
+    });
+  }
+
+  private CompletableFuture<Void> createAllocationTransaction(Budget budget) {
+    Transaction transaction = new Transaction().withAmount(budget.getAllocated())
+        .withFiscalYearId(budget.getFiscalYearId()).withToFundId(budget.getFundId())
+        .withTransactionType(Transaction.TransactionType.ALLOCATION).withSource(Source.USER);
+
+    return handleGetRequest(resourceByIdPath(FISCAL_YEARS, budget.getFiscalYearId(), lang)).
+        thenApply(json -> json.mapTo(FiscalYear.class)).thenAccept(fy ->
+        handleCreateRequest(resourcesPath(TRANSACTIONS), transaction.withCurrency(fy.getCurrency())));
```

Comment:
We discussed at the meeting that the UI should handle the case when the budget is created, but the transaction failed. I know this is not part of the requirements, but could you return a response with a specific code so that the UI can recognize this error?

Reply:
Right, added it.